session/session.go

```diff
@@ -0,0 +1,144 @@
+package session
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/log"
+)
+
+// PortalData ...
+type PortalData struct {
+	AppleID              string              `json:"apple_id"`
+	Password             string              `json:"password"`
+	ConnectionExpiryDate string              `json:"connection_expiry_date"`
+	SessionCookies       map[string][]Cookie `json:"session_cookies"`
+}
+
+// Cookie ...
+type Cookie struct {
+	Name      string `json:"name"`
+	Path      string `json:"path"`
+	Value     string `json:"value"`
+	Domain    string `json:"domain"`
+	Secure    bool   `json:"secure"`
+	Expires   string `json:"expires,omitempty"`
+	MaxAge    int    `json:"max_age,omitempty"`
+	Httponly  bool   `json:"httponly"`
+	ForDomain *bool  `json:"for_domain,omitempty"`
+}
+
+const cookieTemplate = `- !ruby/object:HTTP::Cookie
+  name: <NAME>
+  value: <VALUE>
+  domain: <DOMAIN>
+  for_domain: <FOR_DOMAIN>
+  path: "<PATH>"
+`
+
+// GetSession will fetch the session from Bitrise for the connected Apple developer account
+// If the BITRISE_PORTAL_DATA_JSON is provided (for debug purposes) it will use that instead.
+func GetSession() (string, error) {
+	portalData, err := getDeveloperPortalData(os.Getenv("BITRISE_BUILD_URL"), os.Getenv("BITRISE_BUILD_API_TOKEN"))
+	if err != nil {
+		return "", err
+	}
+
+	cookies := convertDesCookie(portalData.SessionCookies["https://idmsa.apple.com"])
+	session := strings.Join(cookies, "")
+	return session, nil
+}
+
+func getDeveloperPortalData(buildURL, buildAPIToken string) (PortalData, error) {
+	var developerPortalData PortalData
+
+	portalDataJSON, exists := os.LookupEnv("BITRISE_PORTAL_DATA_JSON")
+	if exists && portalDataJSON != "" {
+		return developerPortalData, json.Unmarshal([]byte(portalDataJSON), &developerPortalData)
+	}
+
+	if buildURL == "" {
+		return PortalData{}, fmt.Errorf("BITRISE_BUILD_URL env is not exported")
+	}
+
+	if buildAPIToken == "" {
+		return PortalData{}, fmt.Errorf("BITRISE_BUILD_API_TOKEN env is not exported")
+	}
+
+	req, err := http.NewRequest(http.MethodGet, fmt.Sprintf("%s/apple_developer_portal_data.json", buildURL), nil)
+	if err != nil {
+		return PortalData{}, err
+	}
+
+	req.Header.Add("BUILD_API_TOKEN", buildAPIToken)
+
+	if _, err := performRequest(req, &developerPortalData); err != nil {
+		return PortalData{}, fmt.Errorf("Falied to fetch portal data from Bitrise, error: %s", err)
+	}
+	return developerPortalData, nil
+}
+
+func convertDesCookie(cookies []Cookie) []string {
+	var convertedCookies []string
+	for _, c := range cookies {
```

Comment:
Could use cookie instead of c

Reply:
the type was renamed to cookie from Cookie.